api/v1alpha1/project_config_types.go

```diff
@@ -94,10 +94,15 @@ type WebhookReceiverConfig struct {
 	// GitHub contains the configuration for a webhook receiver that is compatible
 	// with GitHub payloads.
 	//
-	// TODO(fuskovic): Make this mutually exclusive with configs for other platforms.
-	//
-	// +kubebuilder:validation:Required
+	// TODO(fuskovic): Make this mutually exclusive with configs for other
+	// platforms.
 	GitHub *GitHubWebhookReceiverConfig `json:"github,omitempty" protobuf:"bytes,2,opt,name=github"`
+	// GitLab contains the configuration for a webhook receiver that is compatible
+	// with GitLab payloads.
+	//
+	// TODO(fuskovic): Make this mutually exclusive with configs for other
+	// platforms.
+	GitLab *GitHubWebhookReceiverConfig `json:"gitlab,omitempty" protobuf:"bytes,3,opt,name=gitlab"`
```

Comment:
@fuskovic can you work out the mutual exclusivity of these two field, ideally using CEL?